--- a/drivers/net/ethernet/intel/e1000e/netdev.c	2023-11-12 17:23:24.354032818 +0100
+++ b/drivers/net/ethernet/intel/e1000e/netdev.c	2023-11-12 17:31:46.807929904 +0100
@@ -7275,6 +7275,12 @@
 	if ((hw->mac.type >= e1000_pch2lan) && (netdev->mtu > ETH_DATA_LEN))
 		features &= ~NETIF_F_RXFCS;
 
+	/* Fix for hang and reset when forwarding traffic to a client behind NAT,
+	 * which affects some e1000e devices that have PCH2 device */
+	e_info("Disabling TSO to avoid hardware unit hang.\n");
+	features &= ~NETIF_F_TSO;
+	features &= ~NETIF_F_TSO6;
+
 	/* Since there is no support for separate Rx/Tx vlan accel
 	 * enable/disable make sure Tx flag is always in same state as Rx.
 	 */
